sources/packages/api/src/route-controllers/student/student.aest.controller.ts

```diff
@@ -217,14 +219,19 @@ export class StudentAESTController extends BaseController {
   @ApiNotFoundResponse({ description: "Not able to find the student." })
   async getStudentProfile(
     @Param("studentId") studentId: number,
-  ): Promise<StudentProfileAPIOutDTO> {
+  ): Promise<AESTStudentProfileAPIOutDTO> {
     const student = await this.studentControllerService.getStudentProfile(
       studentId,
     );
     if (!student) {
       throw new NotFoundException("Not able to find the student.");
     }
-    return student;
+    const studentRestrictions =
```

Comment:
Because the majority of the time the `NotFoundException` will not be triggered I would say that it is worth making both calls in parallel.